resolwe/flow/models.py

```diff
@@ -373,6 +393,47 @@ def save(self, *args, **kwargs):
 
         super(Data, self).save(*args, **kwargs)
 
+        if created:
+            # Add object to flow_collection:
+            # - only add `Data object` to `flow collection` if process
+            #   has defined `flow_collwection` field
+            # - add object to existing `flow collection` if all `input
+            #   objects`, that belong to `flow collection` (but not
+            #   necessary all `input objects`), belong to the same one
+            # - if `input objects` belong to different `flow
+            #   collections` or don't belong to any `flow collection`,
+            #   create new one
+            if self.process.flow_collection:
+
+                # collect id's of all `input objects`
+                input_objects = []
+                for field_schema, fields, path in iterate_schema(self.input, self.process.input_schema, ''):
+                    if ('type' in field_schema and (
+                            field_schema['type'].startswith('data:') or
+                            field_schema['type'].startswith('list:data:'))):
+                        input_objects.append(fields[field_schema['name']])
+
+                collection_query = Collection.objects.filter(
+                    descriptor_schema__slug=self.process.flow_collection, data__id__in=input_objects).distinct()
+
+                if collection_query.count() == 1:
+                    collection = collection_query.first()
+                else:
+                    des_schema = DescriptorSchema.objects.get(slug=self.process.flow_collection)
+
+                    # XXX: generate meaningful name
```

Comment:
If not pushed explicitly, names should be set by the process. The bioinformatician knows what is meaningful, and it will be different for each process.